x-pack/plugin/ml/src/internalClusterTest/java/org/elasticsearch/xpack/ml/integration/TrainedModelProviderIT.java

```diff
@@ -263,34 +268,39 @@ public void testGetTruncatedModelDefinition() throws Exception {
         assertThat(putConfigHolder.get(), is(true));
         assertThat(exceptionHolder.get(), is(nullValue()));
 
-        List<String> chunks = chunkStringWithSize(config.getCompressedDefinition(), config.getCompressedDefinition().length()/3);
+        // The model definition has been put with the config above but it
+        // is not large enough to be split into chunk. Chunks are required
+        // for this test so overwrite the definition with multiple chunks
+        List<TrainedModelDefinitionDoc.Builder> docBuilders = createModelDefinitionDocs(config.getCompressedDefinition(), modelId);
 
-        List<TrainedModelDefinitionDoc.Builder> docBuilders = IntStream.range(0, chunks.size())
-            .mapToObj(i -> new TrainedModelDefinitionDoc.Builder()
-                .setDocNum(i)
-                .setCompressedString(chunks.get(i))
-                .setCompressionVersion(TrainedModelConfig.CURRENT_DEFINITION_COMPRESSION_VERSION)
-                .setDefinitionLength(chunks.get(i).length())
-                .setEos(i == chunks.size() - 1)
-                .setModelId(modelId))
-            .collect(Collectors.toList());
         boolean missingEos = randomBoolean();
-        docBuilders.get(docBuilders.size() - 1).setEos(missingEos == false);
-        for (int i = missingEos ? 0 : 1 ; i < docBuilders.size(); ++i) {
+        if (missingEos) {
+            // Set the wrong end of stream value
+            docBuilders.get(docBuilders.size() - 1).setEos(false);
+        } else {
+            // else write fewer than the expected number of docs
+            docBuilders.remove(docBuilders.size() -1);
+        }
+        BulkRequestBuilder bulkRequestBuilder = client().prepareBulk();
+        for (int i = 0; i < docBuilders.size(); ++i) {
             TrainedModelDefinitionDoc doc = docBuilders.get(i).build();
-            try(XContentBuilder xContentBuilder = doc.toXContent(XContentFactory.jsonBuilder(),
-                new ToXContent.MapParams(Collections.singletonMap(FOR_INTERNAL_STORAGE, "true")))) {
-                AtomicReference<IndexResponse> putDocHolder = new AtomicReference<>();
-                blockingCall(listener -> client().prepareIndex(InferenceIndexConstants.LATEST_INDEX_NAME)
-                        .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
-                        .setSource(xContentBuilder)
-                        .setId(TrainedModelDefinitionDoc.docId(modelId, 0))
```

Comment:
This test was overwriting the same document rather than writing multiple docs. The code it was testing is sound but fixing this meant changing the test logic